### PR TITLE
Fixing return code error check.

### PR DESCRIPTION
**Pull request by haribert**
Previously the returncode of the if statement was returned rather then
the returncode of the command. Removed unnecessary for loop.

### CODE REVIEW

**Comment by davidtgoldblatt** [comment]
This never actually runs autoconf, right?

I don't actually think we gain anything from returning the error code of the failing command; a script calling autogen.sh shouldn't try to guess the failing subcommand; it just needs a binary success/fail signal.

**Comment by haribert** [comment]
`autogen.sh` does not execute autoconf, but returns the path where it is installed. Or the path where it could not find it.

The return code is only additional information, can be useful in some situation. Otherwise it can just be ignored, and check if it is greater then zero (fail signal).

**Comment by davidtgoldblatt** [comment]
What I mean is, this script will always execute incorrectly, since it never generates the configure script that it wants to execute.

Is there a "root cause" issue you're trying to resolve (e.g. a bug or a desired feature or something)?

**Comment by haribert** [comment]
My commit is perhaps a little bit misleading.

My root cause was the line

> echo "Error $? in ./configure"

which returned "`Error 0 in ./configure`" (because the $? references to the if statement above) and is confusing as an error message. Therefore I introduced the variable `$returncode` and decided to tidy up the rest of the script. But if you want to leave `exit 1` for consistency reasons, feel free to change it back.

File: autogen.sh [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
+fi
 
 echo "./configure --enable-autogen $@"
 ./configure --enable-autogen $@
-if [ $? -ne 0 ]; then
-    echo "Error $? in ./configure"
-    exit 1
+returncode=$?
+if [ $returncode -ne 0 ]; then
+    echo "Error $returncode in ./confinure"
```
Comment (djwatson) [inline autogen.sh]: spelling

**Comment by interwq** [comment]
@haribert: can you please squash the 2 commits into 1? It makes the history a bit cleaner.

**Comment by davidtgoldblatt** [comment]
Two things:
- As Qi mentioned but it seems deleted, the squash
- Let's actually run autoconf instead of just printing the path (i.e. no behavior changes for this script).

**Comment by interwq** [comment]
Closing as no recent update. Please feel free to reopen if needed.

**Pull request closed without merging**